src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStorage.java

```diff
@@ -116,26 +111,43 @@ state to record unknown age.
     }
   }
 
+  /**
+   * Default constructor for external volume
```

Comment:
Hmm this looks odd, having different constructors for different flows when we already are trying to abstract away the differences with the StorageManager interface.
Am guessing what's happening here is that in iceberg mode, we first get the fileLocation from open_channel but subsequently call channel_configure for token renewal. Whereas for iceberg mode we get file location and renewal tokens from the same CLIENT_CONFIGURE api call, thus you're doing it this way? If so I have a suggestion here:
1. As suggested in another thread - ExternalVolumemanager's [line 93](https://github.com/snowflakedb/snowflake-ingest-java/pull/783#discussion_r1653348767) - the StreamingIngestStorage class should just take in an object that knows how to renew tokens and doesn't impose client_configure / channel_configure should have the same call handler.
2. Always take in a FileLocationInfo object for both iceberg and non-iceberg modes, so that the "boostrap" codepath is also identical

this means for non-iceberg, you'll have to make an explicit call to client_configure at the callsite of StreamingIngestStorage.ctor to get the FileLocationInfo for the internal stage, which IMO is okay.

Reply:
Refactored to the same ctor.